src/libs/kata-sys-util/src/mount.rs

```
@@ -0,0 +1,1029 @@
+        "suid" => flags &= !MsFlags::MS_NOSUID,
+        "bind" => flags |= MsFlags::MS_BIND,
+        "dirsync" => flags |= MsFlags::MS_DIRSYNC,
+        "mand" => flags |= MsFlags::MS_MANDLOCK,
+        "noatime" => flags |= MsFlags::MS_NOATIME,
+        "nodev" => flags |= MsFlags::MS_NODEV,
+        "nodiratime" => flags |= MsFlags::MS_NODIRATIME,
+        "noexec" => flags |= MsFlags::MS_NOEXEC,
+        "nosuid" => flags |= MsFlags::MS_NOSUID,
+        "rbind" => flags |= MsFlags::MS_BIND | MsFlags::MS_REC,
+        "relatime" => flags |= MsFlags::MS_RELATIME,
+        "remount" => flags |= MsFlags::MS_REMOUNT,
+        "ro" => flags |= MsFlags::MS_RDONLY,
+        "strictatime" => flags |= MsFlags::MS_STRICTATIME,
+        "sync" => flags |= MsFlags::MS_SYNCHRONOUS,
+        _ => return None,
+    }
+    Some(flags)
+}
+
+// Do mount, optionally change current working directory if `chdir` is not empty.
+fn mount_at<P: AsRef<Path>>(
+    chdir: Option<PathBuf>,
+    source: P,
+    target: PathBuf,
+    fstype: &str,
+    flags: MsFlags,
+    data: &str,
+) -> Result<()> {
+    let chdir = match chdir {
+        Some(v) => v,
+        None => {
+            return mount(
+                Some(source.as_ref()),
+                &target,
+                Some(fstype),
+                flags,
+                Some(data),
+            )
+            .map_err(|e| Error::Mount(PathBuf::new(), target, e));
+        }
+    };
+
+    info!(
+        sl!(),
+        "mount_at: chdir {}, source {}, target {} , fstype {}, data {}",
+        chdir.display(),
+        source.as_ref().display(),
+        target.display(),
+        fstype,
+        data
+    );
+
+    // TODO: enhance security by `secure_path`.
+    let o_flags = nix::fcntl::OFlag::O_PATH | nix::fcntl::OFlag::O_CLOEXEC;
+    let file = fs::OpenOptions::new()
+        .read(true)
+        .custom_flags(o_flags.bits())
+        .open(&chdir)
+        .map_err(|_| Error::InvalidPath(chdir.to_path_buf()))?;
```

Looks like we're losing detail here as the error is discarded, which could make debugging harder.

---

Fixed by introduce error code "OpenByPath".